File: new.pl
#!/usr/bin/env perl
use strict;
use warnings;
use POSIX qw(strftime);

my @date = gmtime;

print "What is the title of the post? ";
my $title = <STDIN>;
chomp $title;
return unless ($title);
my $slug         = lc($title =~ s/ /-/gr);
my $file         = strftime "_posts/%Y/%Y-%m-%d-$slug.md", @date;
my $fm_date      = strftime "%Y-%m-%d %H:%M:%S", @date;
my $front_matter = <<HERE;
---
layout: post
title: $title
date: $fm_date
---

HERE

open(my $fh, '>', $file) or die "Could not write file $file $!";
print $fh $front_matter;
close($fh);

File: new.rb
#!/usr/bin/env ruby

require('time')
now = Time.now

print 'What is the title of the post? '
title = gets.strip
return if title.empty?

slug = title.downcase.tr(' ', '-')
year = now.strftime('%Y')
folder = "_posts/#{year}"
Dir.mkdir(folder) unless Dir.exist?(folder)
filename = "#{folder}/#{now.strftime('%Y-%m-%d')}-#{slug}.md"
fm_date = now.strftime('%Y-%m-%d %H:%M:%S')
front_matter = <<~HERE
  ---
  layout: post
  title: #{title}
  date: #{fm_date}
  ---
HERE

File.open(filename, 'w') { |f| f.write(front_matter) }
